src/server/models/list.ts

```diff
@@ -250,3 +250,42 @@ export async function updateAnnualReviewDate(listId: string, nextAnnualReviewSta
     },
   });
 }
+
+export async function findListDashboardData(listId: string) {
+  return await prisma.listsForDashboard.findUnique({
+    where: {
+      listId: Number(listId),
+    },
+  });
+}
+
+export async function deleteList(list: List, userEmail: string) {
+  await prisma.$transaction([
+    prisma.listItem.deleteMany({
+      where: {
+        listId: Number(list.id),
+      },
+    }),
+    prisma.list.delete({
+      where: {
+        id: Number(list.id),
+      },
+    }),
```

Comment:
I think you can delete list items and lists in one go. but you will need to change `onCascade` See https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions

I think probably `Address` would have to be deleted too if they're linked to `listItem` (might just be deleted automatically though - worth checking).

Reply:
Yep, looks like addresses weren't getting deleted. I wasn't able to get cascading deletes to work with this so I've had to write a nasty interactive transaction.

Let me know what you think.